packages/preact/src/index.ts

```diff
@@ -80,8 +80,13 @@ function Text(this: AugmentedComponent, { data }: { data: Signal }) {
 			}
 		}
 
-		// Replace this component's vdom updater with a direct text one:
 		this._updater!._callback = () => {
+			if (isValidElement(s.peek()) || this.base?.nodeType !== 3) {
```

Comment:
`isValidElement` would trigger for states that are created by means of `createElement('hi')` which is a valid textNode that we could perform the optimised render on so it will be safer to just check for `this.base.nodeType`

```suggestion
			if (this.base?.nodeType !== 3) {
```

Reply:
createElement('hi') will produce this -> `<hi></hi>`, is not it?

Reply:
Ah true internally we started using createVNode for this, in Preact a nullish type can be an individual text node hence the commwnt